### PR TITLE
Fix topics storage load strange data

### PR DESCRIPTION
**Pull request by kstonekuan**
Make sure topic title and question description cannot be blank or duplicate. Allow other data (hint, explanation, option) to be blank or duplicate in case user wishes to do this for some reason.

### CODE REVIEW

File: src/main/java/seedu/eduke8/storage/TopicsStorage.java [hunk 1]
```diff
@@ -15,25 +15,31 @@
...
+import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TITLE_BLANK;
+import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TITLE_DUPLICATE;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TOO_MANY_CORRECT;
 import static seedu.eduke8.exception.ExceptionMessages.ERROR_TOPICS_JSON_TOPIC;
 
 public class TopicsStorage extends LocalStorage {
     private boolean wasCorrectAnswerMarked;
     private String currentQuestionDescription;
     private String currentTopicTitle;
+    private final HashSet<String> topicTitles;
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/storage/TopicsStorage.java:34]: Great implementation to check for duplicated topics! Do you think the same should be done for questions?

File: src/main/java/seedu/eduke8/storage/TopicsStorage.java [hunk 2]
```diff
@@ -82,6 +100,12 @@ private Topic parseToTopicObject(JSONObject topic) throws Eduke8Exception {
 
     private Question parseToQuestionObject(JSONObject question) throws Eduke8Exception {
         currentQuestionDescription = (String) question.get(KEY_DESCRIPTION);
+
+        if (currentQuestionDescription.equals("")) {
+            throw new Eduke8Exception(ERROR_TOPICS_JSON_PREFACE
+                    + System.lineSeparator() + ERROR_TOPICS_JSON_DESCRIPTION_BLANK);
+        }
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/storage/TopicsStorage.java]: Great that you are checking for empty titles and throwing an exception error to the user!

**Review by Zhi-You — COMMENTED** [review]
LGTM! just wondering if duplicated questions can be handled too!

File: src/main/java/seedu/eduke8/storage/TopicsStorage.java (hunk 1 above)
Comment (kstonekuan) [reply src/main/java/seedu/eduke8/storage/TopicsStorage.java:34]: Implemented this as well

Comment (Zhi-You) [reply src/main/java/seedu/eduke8/storage/TopicsStorage.java:34]: thanks!

**Review by Zhi-You — APPROVED** [review]
LGTM !

**Pull request merged**